src/services/review/RepoCheckerService.ts

```diff
@@ -433,16 +459,24 @@ export default class RepoCheckerService {
       fs.writeFileSync(filePath, stringifiedErrors)
     }
 
+    let gitOperations:
+      | Response<CommitResult>
+      | Response<PushResult> = this.git
+      .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
+      .checkout("main")
+      .fetch()
+      .merge(["origin/main"])
+      .add(["."])
+      .commit(`Site checker logs added for ${repo}`)
+
+    const isProd = config.get("env") === "prod"
+    if (isProd) {
+      gitOperations = gitOperations.push()
+    }
+
     // pushing since there is no real time requirement for user
-    return ResultAsync.fromPromise(
-      this.git
-        .cwd({ path: SITE_CHECKER_REPO_PATH, root: false })
-        .checkout("main")
-        .fetch()
-        .merge(["origin/main"])
-        .add(["."])
-        .commit(`Site checker logs added for ${repo}`)
-        .push(),
+    return ResultAsync.fromPromise<CommitResult | PushResult, SiteCheckerError>(
+      gitOperations,
```

Comment:
can we put the whole operation into the `if isProd` branch

Reply:
hey may i clarify if this is a nit ya? this would be useful for testing on staging branch, just without pushing to the remote repository so that only prod uses it. happy to huddle to explain this concept over ya

Reply:
hmm, for me, i think the problem here is that we `commit` on non-prod environments and don't do anything with it. in that case, why bother committing in the first place? 

ie, we do a `commit + push` only `if isProd`, which makes it clearer

Reply:
sure [5cb14f5](https://github.com/isomerpages/isomercms-backend/pull/1187/commits/5cb14f5edd3233d524e4014177680b4c9957a33f)